src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java

```diff
@@ -29,33 +32,44 @@
 import static org.junit.jupiter.api.Assertions.fail;
 
 @DatabaseTest
+@Execution(ExecutionMode.SAME_THREAD)
 class DBMSProcessorTest {
 
-    @ParameterizedTest
-    @MethodSource("org.jabref.logic.shared.TestManager#getTestingDatabaseSystems")
-    void testCheckBaseIntegrity(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException {
+    private DBMSConnection dbmsConnection;
+    private DBMSProcessor dbmsProcessor;
+    private DBMSType dbmsType;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        this.dbmsType = TestManager.getDBMSTypeTestParameter();
+        this.dbmsConnection = TestConnector.getTestDBMSConnection(dbmsType);
+        this.dbmsProcessor = DBMSProcessor.getProcessorInstance(TestConnector.getTestDBMSConnection(dbmsType));
+        TestManager.clearTables(this.dbmsConnection);
+    }
+
+    @AfterEach
+    public void clear() throws SQLException {
+        this.dbmsConnection.getConnection().close();
+    }
+
+    @Test
+    void testCheckBaseIntegrity() throws SQLException {
         dbmsProcessor.setupSharedDatabase();
         assertTrue(dbmsProcessor.checkBaseIntegrity());
-        clear(dbmsConnection);
+        TestManager.clearTables(this.dbmsConnection);
         assertFalse(dbmsProcessor.checkBaseIntegrity());
-
-        clear(dbmsConnection);
     }
 
-    @ParameterizedTest
-    @MethodSource("org.jabref.logic.shared.TestManager#getTestingDatabaseSystems")
-    void testSetUpSharedDatabase(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException {
+    @Test
+    void testSetUpSharedDatabase() throws SQLException {
         dbmsProcessor.setupSharedDatabase();
-        clear(dbmsConnection);
+        clear();
```

Comment:
Is the `clear` really necessary? If I understand the code correctly, then it tests that running `setup` twice results in a correct db.

Reply:
I split the tests, duplicated, modified and added `databaseIntegrityFullFiledAfterClearAndSetup`. Just to ensure that "setup" also works on a clear database.

The double call test is removed.